src/main/java/seedu/address/logic/commands/DeleteOrderCommand.java

```diff
@@ -37,14 +38,35 @@ public DeleteOrderCommand(Index targetIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Order> lastShownList = model.getFilteredOrderList();
 
+        Order orderToDelete = getOrderToDelete(model);
+        cascadeDeleteCheeses(model, orderToDelete);
+        model.deleteOrder(orderToDelete);
+        model.setPanelToOrderList(); // Display order list
+
+        // Reset to full customer list so as to create the order list panel correctly
+        model.updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
+
+        return new CommandResult(String.format(MESSAGE_DELETE_ORDER_SUCCESS, orderToDelete));
+    }
+
+    /**
+     * Obtains the {@code Order} to delete using the target index.
+     */
+    private Order getOrderToDelete(Model model) throws CommandException {
+        List<Order> lastShownList = model.getFilteredOrderList();
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
         }
+        return lastShownList.get(targetIndex.getZeroBased());
+    }
 
-        Order orderToDelete = lastShownList.get(targetIndex.getZeroBased());
+    /**
+     * Cascade-delete the cheeses assigned to the order that is being deleted.
+     */
+    private void cascadeDeleteCheeses(Model model, Order orderToDelete) throws CommandException {
         model.updateCheesesStatus(orderToDelete.getCheeses());
+        model.updateFilteredCheeseList(PREDICATE_SHOW_ALL_CHEESES);
```

Comment:
Same as the comment in #37, don't think that we should change the UI portion just to delete the cheeses (although functionaility-wise it is exactly the same, just the organization is a bit different)